packages/snaps-controllers/src/websocket/WebSocketService.ts

```diff
@@ -0,0 +1,254 @@
+import type { RestrictedMessenger } from '@metamask/base-controller';
+import { rpcErrors } from '@metamask/rpc-errors';
+import type { SnapId, WebSocketEvent } from '@metamask/snaps-sdk';
+import { HandlerType, logError } from '@metamask/snaps-utils';
+import { assert, createDeferredPromise } from '@metamask/utils';
+import { nanoid } from 'nanoid';
+
+import type {
+  HandleSnapRequest,
+  SnapInstalled,
+  SnapUninstalled,
+  SnapUpdated,
+} from '../snaps';
+import { METAMASK_ORIGIN } from '../snaps';
+
+const serviceName = 'WebSocketService';
+
+export type OpenWebSocket = {
```

Comment:
Wondering if we should start naming these more consistently across controllers/services, similar to how most controllers in `MetaMask/core` do it, e.g.,

- `WebSocketServiceOpenAction`
- `WebSocketServiceSendMessageAction`
- etc.